python/ray/train/_internal/stats.py

```diff
@@ -0,0 +1,117 @@
+import logging
+import os
+import threading
+from typing import Dict, Optional
+
+import ray
+from ray.data import Dataset
+from ray.train._internal.schema import TrainDatasetInfo, TrainRunInfo, TrainWorkerInfo
+from ray.train._internal.utils import check_for_failure
+from ray.train._internal.worker_group import WorkerGroup
+
+logger = logging.getLogger(__name__)
+
+
+@ray.remote(num_cpus=0)
+class TrainStatsActor:
+    def __init__(self):
+        self.train_runs = dict()
+
+    def register_train_run(self, run_info: TrainRunInfo):
+        # Register a new train run.
+        self.train_runs[run_info.id] = run_info
+
+    def get_train_run(self, run_id: str) -> Optional[TrainRunInfo]:
+        # Retrieve a registered run with its id
+        return self.train_runs.get(run_id, None)
+
+    def get_all_train_runs(self) -> Dict[str, TrainRunInfo]:
+        # Retrieve all registered train runs
+        return self.train_runs
+
+
+TRAIN_STATS_ACTOR_NAME = "train_stats_actor"
+TRAIN_STATS_ACTOR_NAMESPACE = "_train_stats_actor"
+
+_stats_actor_lock: threading.RLock = threading.RLock()
+
+
+def get_or_launch_stats_actor():
+    """Create or launch a `TrainStatsActor` on the head node."""
+    with _stats_actor_lock:
+        return TrainStatsActor.options(
+            name=TRAIN_STATS_ACTOR_NAME,
+            namespace=TRAIN_STATS_ACTOR_NAMESPACE,
+            get_if_exists=True,
+            lifetime="detached",
+            resources={"node:__internal_head__": 0.001},
+        ).remote()
+
+
+class TrainRunStatsManager:
+    """A class that aggregates and reports train run info to TrainStatsActor.
+
+    This manager class is created on the train controller layer for each run.
+    """
+
+    def __init__(self) -> None:
+        self.stats_actor = get_or_launch_stats_actor()
+
+    def register_train_run(
+        self,
+        run_id: str,
+        run_name: str,
+        trial_name: str,
+        trainer_actor_id: str,
+        datasets: Dict[str, Dataset],
+        worker_group: WorkerGroup,
+    ) -> None:
+        """Collect Train Run Info and report to StatsActor."""
+
+        def collect_train_worker_info():
+            train_context = ray.train.get_context()
```

Comment:
Do we want to tie this to the Ray Train context? We actually have the world / local / node rank info on the trainer already.

Reply:
I am trying to disentangle the dashboard backend with our current Trainer implementation, so that it could be easily migrated to new Trainer in the future.

Reply:
Yeah. I prefer collecting the information directly from workers instead of controller, since that is exactly what the users see, which should be treated as the ground truth.